### PR TITLE
feat: add gio-license directive

### PR DESCRIPTION
**Pull request by gcusnieux**
## Issue

https://gravitee.atlassian.net/browse/APIM-1831

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-molykkobqz.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.directive.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import { GioLicenseService } from './gio-license.service';
+
+import { GioEeUnlockDialogComponent, GioEeUnlockDialogData } from '../../../components/gio-ee-unlock-dialog/gio-ee-unlock-dialog.component';
+import { FeatureMoreInformation } from '../../../entities/feature/FeatureMoreInformation';
+
+export interface GioLicenseOptions {
+  feature?: string;
+}
+
+const featureMoreInformationData = {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/shared/components/gio-license/gio-license.directive.ts]: wdyt to add it into licenseService ?

File: gravitee-apim-console-webui/src/shared/interceptors/replace-env.interceptor.ts [hunk 2]
```diff
@@ -24,6 +24,9 @@ export class ReplaceEnvInterceptor implements HttpInterceptor {
   constructor(@Inject('Constants') private readonly constants: Constants) {}
 
   intercept(req: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    if (req.url.indexOf('{:envId}') === -1) {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/shared/interceptors/replace-env.interceptor.ts:27]: what about adding a test for this if ?

**Review by ThibaudAV — APPROVED** [review]
I'm always afraid of this kind of parasitic feature added almost everywhere ^^.

**Review by a-cordier — CHANGES_REQUESTED** [review]
Just needs renaming the features as below. 

```java
String FEATURE_CUSTOM_ROLES = "apim-custom-roles";
String FEATURE_DEBUG_MODE = "apim-debug-mode";
String FEATURE_SHARDING_TAGS = "apim-sharding-tags";
String FEATURE_OPEN_ID_CONNECT_SSO = "apim-openid-connect-sso";
String FEATURE_AUDIT_TRAIL = "api-audit-trail";
String FEATURE_DCR_REGISTRATION = "apim-dcr-registration";
```

**Comment by a-cordier** [comment]
Regarding the mockup, I think here the lock should be on audit and not on history and events

Screenshot 2023-06-15 at 15 00 14

**Comment by a-cordier** [comment]
I think there should be a lock here as well
Screenshot 2023-06-15 at 15 03 34

As in Organization | Audit when the screen is merged

**Review by a-cordier — CHANGES_REQUESTED** [review]
see comments 👆

**Pull request merged**